java-checks-test-sources/src/main/java/checks/security/AndroidMobileDatabaseEncryptionKeysCheck.java

```diff
@@ -0,0 +1,78 @@
+package checks.security;
+
+import io.realm.RealmConfiguration;
+import java.nio.charset.StandardCharsets;
+import net.sqlcipher.database.SQLiteDatabase;
+import net.sqlcipher.database.SQLiteDatabaseHook;
+
+class AndroidMobileDatabaseEncryptionKeysCheck {
+
+  byte[] a = new byte[0];
+
+  String getPassword() {
+    return null;
+  }
+
+  void SQLiteDatabase_openOrCreateDatabase(String databasePath, SQLiteDatabaseHook hook, SQLiteDatabase.CursorFactory cursorFactory) {
+    String password = "pwd";
+    SQLiteDatabase database;
+
+    database = new SQLiteDatabase("path", "pwd".toCharArray(), cursorFactory, 0); // Noncompliant
+    database = new SQLiteDatabase("path", password.toCharArray(), cursorFactory, 0); // Noncompliant
```

Comment:
```
char[] a = "pwd".toCharArray();
database = new SQLiteDatabase("path", a, cursorFactory, 0);
```
We should add a test for this situation to show that we are also supporting this case.